### PR TITLE
164829372 List Users

### PR DESCRIPTION
**Pull request by LarryKarani**
__What does this PR do?__
Enables authenticated users to get all the available authors and their profiles

__Description of the task to be completed__
Have the following endpoints working

__GET localhost/api/profiles/__
get the profiles of all authors

__GET localhost/api/users/__
get all authors

__How should this be manually tested?__

- Clone the repo and navigate to the root directory
- Create a virtual environment by typing` virtualenv venv` and activate it
- Install requirements by typing `pip install -r dev-requirements.txt`

Setup a database and define the DATABASE_URL in a .env file. Export the configuration variable
Run the tests by typing python `manage.py test`

__Relevant PT stories__
https://www.pivotaltracker.com/story/show/164829372

### CODE REVIEW

**Comment by kihuha** [comment]
Don't you think it would be more appropriate to only have one level when retrieving all users? (What I mean by this is that instead of having to get the `user` dict then the `users` dict, we just have one dict level)
Screenshot 2019-04-09 at 14 24 35

**Comment by LarryKarani** [comment]
> Don't you think it would be more appropriate to only have one level when retrieving all users? (What I mean by this is that instead of having to get the `user` dict then the `users` dict, we just have one dict level)
> Screenshot 2019-04-09 at 14 24 35

Thanks for the feedback. I'm fixing it right away

**Comment by LarryKarani** [comment]
> Don't you think it would be more appropriate to only have one level when retrieving all users? (What I mean by this is that instead of having to get the `user` dict then the `users` dict, we just have one dict level)
> Screenshot 2019-04-09 at 14 24 35

Its now fixed 

Screenshot 2019-04-09 at 14 35 17

**Review by kihuha — DISMISSED** [review]

File: authors/apps/profiles/tests/test_profiles.py [hunk 1]
```diff
@@ -93,6 +93,16 @@ def setUp(self):
         self.token = self.login.data['token']
         self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
```
Comment (WNjihia) [inline authors/apps/profiles/tests/test_profiles.py:95]: We could also test if an authenticated user can get user profiles

**Review by WNjihia — CHANGES_REQUESTED** [review]
Hey some of the files in the list users pr are not related to the task. How come? It's out of scope for this particular pr
authors/apps/authentication/*
authors/apps/core/utils.py

**Comment by WNjihia** [comment]
Also, there are merge conflicts. Kindly fix them

File: authors/apps/profiles/tests/test_profiles.py (hunk 1 above)
Comment (LarryKarani) [reply authors/apps/profiles/tests/test_profiles.py:95]: > We could also test if an authenticated user can get user profiles

This has already been tested in `test_profiles.py` from line 96 to 104

**Comment by LarryKarani** [comment]
> Hey some of the files in the list users pr are not related to the task. How come? It's out of scope for this particular pr
> authors/apps/authentication/*
> authors/apps/core/utils.py

1. Authentication contains user views and tests. I added a method to get all the users in this view.

2. Inside the utils.py file, I added a method that overwrites the permissions classes so that I could specifically add permission to the get view and seclude the signup post method as they both were inside the same view.

**Comment by LarryKarani** [comment]
> Also, there are merge conflicts. Kindly fix them

Okay

**Pull request closed without merging**